tests/flamegraph.rs

```diff
@@ -116,9 +167,128 @@ fn flamegraph_nameattr() {
         hash: true,
         func_frameattrs: flamegraph::FuncFrameAttrsMap::from_file(&PathBuf::from(nameattr_file))
             .unwrap(),
-        title: "Flame Graph".to_string(),
         ..Default::default()
     };
 
-    test_flamegraph(input_file, expected_result_file, options);
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_should_warn_about_fractional_samples() {
+    test_flamegraph_logs(
+        "./tests/data/fractional-samples/fractional.txt",
+        |captured_logs| {
+            assert_eq!(captured_logs.len(), 1);
+            assert!(captured_logs[0].body.contains("fractional sample"));
+            assert_eq!(captured_logs[0].level, Level::Warn);
+        },
+    );
+}
+
+#[test]
+fn flamegraph_should_not_warn_about_zero_fractional_samples() {
+    test_flamegraph_logs(
+        "./tests/data/fractional-samples/zero-fractionals.txt",
+        |captured_logs| {
+            assert!(captured_logs.is_empty());
+        },
+    );
+}
+
+#[test]
+fn flamegraph_palette_map() {
+    let input_file = "./flamegraph/test/results/perf-vertx-stacks-01-collapsed-all.txt";
+    let expected_result_file = "./tests/data/palette-map/consistent-palette.svg";
+    let palette_file = "./tests/data/palette-map/palette.map".to_string();
+
+    let options = flamegraph::Options {
+        consistent_palette: true,
+        palette_file,
+        ..Default::default()
+    };
+
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_should_warn_about_bad_input_lines() {
+    test_flamegraph_logs("./tests/data/bad-lines/bad-lines.txt", |captured_logs| {
+        assert_eq!(captured_logs.len(), 1);
+        assert!(captured_logs[0].body.contains("invalid format"));
+        assert_eq!(captured_logs[0].level, Level::Warn);
```

Comment:
Don't we also print something along the lines of "N lines ignored"? I guess that's only in collapsing.
In any case, I think we should be liberal here too.